### ISSUE TITLE
absolute/fixed positioned elements inside row-detail

### ISSUE THREAD

**Issue by kamil-maslowski** [Saulis/iron-data-table#174]
There's an issue with absolute/fixed elements that are inside `row-detail` as they are overlapped by other list item elements.

It makes problems when e.g. implementing datepicker inside details row.

Please see attached 
![screen shot 2016-11-30 at 12 28 58](https://cloud.githubusercontent.com/assets/1447277/20750757/bff4a95a-b6f8-11e6-9d2a-b28a566f6408.png)

**Comment by web-padawan** [comment]
the reason is that `iron-list` uses `transform` property: https://github.com/PolymerElements/iron-list/issues/242

**Comment by kamil-maslowski** [comment]
As I thought... but is there any legit way to fix that when using `iron-data-table`?

**Comment by web-padawan** [comment]
I used to fix it [like this](https://github.com/PolymerElements/iron-list/issues/242#issuecomment-232450210), but now I ended up with creating [brainy-table](https://github.com/OWOX/brainy-table) which uses `dom-repeat` (and I'm going to implement pagination to deal wit big amounts of data).

**Comment by kamil-maslowski** [comment]
Yeah, found the `z-index` fix myself... but that works only "one way", from top to bottom, and if something will pop to top, earlier elements will overlap over it.

Other than that... how did you managed to set `z-index` on list items?

**Comment by web-padawan** [comment]
@kamil-maslowski here is the solution to calculate `z-index`. Paste this code in `beforeRowBind`:
```
var items = Polymer.dom(table.$$('#list')).querySelectorAll('div.item');
items.forEach(function (item, index) {
    item.style.zIndex = items.length - index;
});
```

**Comment by kamil-maslowski** [comment]
@web-padawan I did that kinda different to overcome the problem with previous rows overlapping elements:

`_beforeRowBind: function(data, row) {
row.parentNode.style.zIndex = data.expanded ? "1" : null;
},`

just fyi ;)

**Comment by web-padawan** [comment]
@kamil-maslowski please take into account that using `?dom=shadow` (or enabling native shadow DOM in global Polymer settings) changes table structure. There is a [trick](https://github.com/Saulis/iron-data-table/blob/master/data-table-row.html#L88) with `<content>` element and node distribution :)

**Comment by anooprajan89** [comment]
**@web-padawan**  I tried your code. 

> var items = Polymer.dom(table.$$('#list')).querySelectorAll('div.item');
> items.forEach(function (item, index) {
>     item.style.zIndex = items.length - index;
> });

But the issue coming opposite. In my issue, the calendar shows upward is fine. Downward is behind the column. After using your code, It's coming opposite(downword is fine, but upward is going behind table).
Can you figure it out?
